File: tusow/src/main/java/it/unibo/coordination/tusow/api/TextualTupleSpaceApi.java
package it.unibo.coordination.tusow.api;

import io.vertx.ext.web.RoutingContext;
import it.unibo.coordination.linda.text.RegexTemplate;
import it.unibo.coordination.linda.text.RegularMatch;
import it.unibo.coordination.linda.text.StringTuple;

public interface TextualTupleSpaceApi extends TupleSpaceApi<StringTuple, RegexTemplate, Object, String, RegularMatch> {
    static TextualTupleSpaceApi get(RoutingContext context) {
        return new TextualTupleSpaceApiImpl(context);
    }
}

File: tusow-service/src/main/kotlin/it/unibo/coordination/tusow/api/TextualTupleSpaceApi.kt
package it.unibo.coordination.tusow.api

import io.vertx.ext.web.RoutingContext
import it.unibo.coordination.linda.text.RegexTemplate
import it.unibo.coordination.linda.text.RegularMatch
import it.unibo.coordination.linda.text.StringTuple

interface TextualTupleSpaceApi : TupleSpaceApi<StringTuple, RegexTemplate, Any, String, RegularMatch> {
    companion object {
        @JvmStatic
        operator fun get(context: RoutingContext): TextualTupleSpaceApi {
            return TextualTupleSpaceApiImpl(context)
        }
    }
}